plugins/provisioners/puppet/config/puppet.rb

```diff
@@ -85,6 +87,9 @@ def finalize!
             if @manifest_file == UNSET_VALUE
               @manifest_file = nil
             end
+            if @environment_variables == UNSET_VALUE
+              @environment_variables = nil
```

Comment:
I think, if you set this to `{}`, you'll have a better day. Then you don't need any nil checks - you can just .each it all the way 🏡